front/src/applications/stdcmV2/hooks/useStaticPathfinding.ts

```diff
@@ -3,17 +3,19 @@ import { useEffect, useMemo, useState } from 'react';
 import { compact } from 'lodash';
 import { useSelector } from 'react-redux';
 
-import { osrdEditoastApi, type PathfindingResult } from 'common/api/osrdEditoastApi';
+import {
+  osrdEditoastApi,
+  type InfraWithState,
+  type PathfindingResult,
+} from 'common/api/osrdEditoastApi';
 import { useOsrdConfSelectors } from 'common/osrdContext';
-import useInfraStatus from 'modules/pathfinding/hooks/useInfraStatus';
 import usePathProperties from 'modules/pathfinding/hooks/usePathProperties';
 import { getPathfindingQuery } from 'modules/pathfinding/utils';
 import { useStoreDataForRollingStockSelector } from 'modules/rollingStock/components/RollingStockSelector/useStoreDataForRollingStockSelector';
 
-const useStaticPathfinding = () => {
+const useStaticPathfinding = (infra?: InfraWithState) => {
```

Comment:
instead of passing on optional param, shouldn't we check `infra` before using `useStaticPathfinding`?

Reply:
I can not condition the execution of a hook (i.e. sometimes execute it and sometimes not, otherwise we will have the error: "not the same number of hooks"), so we have to pass the infra even if it is undefined and the test must be done in the hook ;)